### PR TITLE
[Feature]: Manage Select options Via interface

### PR DESCRIPTION
**Pull request by aryaantony92**
Refactored admin-ui bundle changes from https://github.com/pimcore/pimcore/pull/14454

### CODE REVIEW

**Comment by kjkooistra-youwe** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/OptionsProvider/SelectOptionsOptionsProvider.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace Pimcore\Bundle\AdminBundle\OptionsProvider;
+
+use Pimcore\Model\DataObject\ClassDefinition\Data;
+use Pimcore\Model\DataObject\ClassDefinition\DynamicOptionsProvider\SelectOptionsProviderInterface;
+use Pimcore\Model\DataObject\SelectOptions\Config;
+use Pimcore\Model\DataObject\SelectOptions\Data\SelectOption;
+use Pimcore\Model\DataObject\Service;
+
+class SelectOptionsOptionsProvider implements SelectOptionsProviderInterface
```
Comment (dvesh3) [inline src/OptionsProvider/SelectOptionsOptionsProvider.php]: @aryaantony92 why implemented here not in the core?

File: public/js/pimcore/object/helpers/selectField.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            hidden: true,
+            valueField: 'id',
+            displayField: 'text',
+            editable: false,
+            forceSelection: true,
+            queryMode: 'local',
+            store: this.getSelectOptionsStore(),
+            listeners: {
+                change: function (comboBox, newValue) {
+                    optionsProviderClass.setValue('\\Pimcore\\Bundle\\AdminBundle\\OptionsProvider\\SelectOptionsOptionsProvider');
```
Comment (kjkooistra-youwe) [inline public/js/pimcore/object/helpers/selectField.js]: This will have to be updated to CoreBundle

Comment (aryaantony92) [inline public/js/pimcore/object/helpers/selectField.js]:
```suggestion
                    optionsProviderClass.setValue('\\Pimcore\\Bundle\\CoreBundle\\OptionsProvider\\SelectOptionsOptionsProvider');
```

File: public/js/pimcore/object/helpers/selectField.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            hidden: true,
+            valueField: 'id',
+            displayField: 'text',
+            editable: false,
+            forceSelection: true,
+            queryMode: 'local',
+            store: this.getSelectOptionsStore(),
+            listeners: {
+                change: function (comboBox, newValue) {
+                    optionsProviderClass.setValue('\\Pimcore\\Bundle\\CoreBundle\\OptionsProvider\\SelectOptionsOptionsProvider');
```
Comment (dvesh3) [inline public/js/pimcore/object/helpers/selectField.js]: can we handle this with `pimcore.settings.select_options_provider_class`? https://github.com/pimcore/admin-ui-classic-bundle/blob/42ae0d40aa43d163e1bde2aa46f623830a406063/src/Controller/Admin/IndexController.php#L293

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**